### PR TITLE
Matched some world/area_kkj functions

### PR DESCRIPTION
**Pull request by mrshigure**

### CODE REVIEW

File: src/world/area_hos/hos_00/A0BCF0.c [hunk 1]
```diff
@@ -4,7 +4,7 @@ extern char D_800D9668[];
 
 ApiStatus func_80240000_A0BCF0(Evt* script, s32 isInitialCall) {
     if (gGameStatusPtr->entryID == 3) {
-        sprintf(&D_800D9668, "hos_bg\0\0\0\0\0\0\0\0\0");
+        sprintf(D_800D9668, "hos_bg\0\0\0\0\0\0\0\0\0");
```
Comment (mrshigure) [inline src/world/area_hos/hos_00/A0BCF0.c:7]: Updating this to remove a warning.

**Comment by BowserSlug** [comment]
✅ This PR fixes 1 warnings!

File: src/world/area_kkj/kkj_25/B06D10.c [hunk 2]
```diff
@@ -1,35 +1,176 @@
...
+#include "effects.h"
 
 #include "world/common/UnkFoldFunc.inc.c"
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_25/B06D10", func_802406EC_B070EC);
+ApiStatus func_802406EC_B070EC(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 spriteIndex = evt_get_variable(script, *args++);
+    s32 rasterIndex = evt_get_variable(script, *args++);
+    UnkEntityStruct* unkStruct = heap_malloc(sizeof(*unkStruct));
```
Comment (mrshigure) [inline src/world/area_kkj/kkj_25/B06D10.c:10]: Based on a similar file and on the struct size, I assumed this to be an `UnkEntityStruct`, but I can't tell for sure. Would it be preferable to just create another `UnkStruct`?

File: src/world/area_kkj/kkj_25/B06D10.c [hunk 3]
```diff
@@ -1,35 +1,176 @@
...
+    fx_stars_orbiting(0, script->varTable[0], script->varTable[1], script->varTable[2], script->varTable[3], script->varTable[4], &effect);
+    script->varTablePtr[0] = effect;
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80240B8C_B0758C(Evt* script, s32 isInitialCall) {
+    EffectInstance* effect = (EffectInstance*) script->varTablePtr[0];
+
+    effect->data.starsOrbiting->unk_28 = 0;
```
Comment (mrshigure) [inline src/world/area_kkj/kkj_25/B06D10.c:109]: Assumed the data to be a `StarsOrbitingFXData` since we have some calls to `fx_stars_orbiting` in this file. Was this the right call or should I revert these changes?

File: src/world/area_kkj/kkj_25/B06D10.c (hunk 2 above)
Comment (nanaian) [reply src/world/area_kkj/kkj_25/B06D10.c:10]: Looking at the fields set below this is almost certainly a correct assumption, I say keep UnkEntityStruct 👍

File: src/world/area_kkj/kkj_25/B06D10.c (hunk 3 above)
Comment (nanaian) [reply src/world/area_kkj/kkj_25/B06D10.c:109]: Lgtm

**Review by nanaian — APPROVED** [review]

File: src/world/area_kkj/kkj_10/ACAB20.c [hunk 4]
```diff
@@ -7,6 +7,19 @@ static char* N(exit_str_3) = "kkj_19";
 static char* N(exit_str_4) = "kkj_20";
 static char* N(exit_str_5) = "kkj_29";
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_10/ACAB20", func_80240000_ACAB20);
+ApiStatus func_80240000_ACAB20(Evt* script, s32 isInitialCall) {
+    s32 i;
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_10/ACAB20", func_80240034_ACAB54);
+    for (i = 0; i < 7; i++) {
```
Comment (ethteck) [inline src/world/area_kkj/kkj_10/ACAB20.c:13]: Instead of `7` here, we tend to do ARRAY_COUNT(thing) instead - so `ARRAY_COUNT(D_80151328->sources)`

File: src/world/area_kkj/kkj_11/ACE140.c [hunk 5]
```diff
@@ -9,6 +9,19 @@ static char* N(exit_str_5) = "kkj_17";
 static char* N(exit_str_6) = "kkj_18";
 static char* N(exit_str_7) = "kkj_11";
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_11/ACE140", func_80240000_ACE140);
+ApiStatus func_80240000_ACE140(Evt* script, s32 isInitialCall) {
+    s32 i;
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_11/ACE140", func_80240034_ACE174);
+    for (i = 0; i < 7; i++) {
```
Comment (ethteck) [inline src/world/area_kkj/kkj_11/ACE140.c:15]: here2

File: src/world/area_kkj/kkj_14/ADAEA0.c [hunk 6]
```diff
@@ -1,3 +1,10 @@
 #include "kkj_14.h"
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_14/ADAEA0", func_80240000_ADAEA0);
+extern char D_800D9668[];
+
+ApiStatus func_80240000_ADAEA0(Evt* script, s32 isInitialCall) {
+    if (evt_get_variable(NULL, GSW(0)) == -128) {
+        sprintf(D_800D9668, "nok_bg\0\0pra_01\0\0kkj_11\0\0kkj_01\0\0kkj_26\0\0kkj_15\0\0kkj_27\0\0kkj_26");
```
Comment (ethteck) [inline src/world/area_kkj/kkj_14/ADAEA0.c:7]: technically this should just be nok_bg and there should be some other static char arrays (or one big one for now) for the rest of these strings. you could place it directly after this function.

File: src/world/area_kkj/kkj_14/ADAEF0.c [hunk 7]
```diff
@@ -1,17 +1,44 @@
...
-#include "world/common/atomic/TexturePan.inc.c"
+extern s32 D_80241D38_ADCBD8;
+extern u8 D_80241D3B_ADCBDB;
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_14/ADAEF0", func_8024035C_ADB1FC);
+#include "world/common/atomic/TexturePan.inc.c"
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_14/ADAEF0", func_80240384_ADB224);
+ApiStatus func_8024035C_ADB1FC(Evt* script, s32 isInitialCall) {
+    gGameStatusPtr->peachFlags &= 0xFE;
```
Comment (ethteck) [inline src/world/area_kkj/kkj_14/ADAEF0.c:9]: &= ~0x1

File: src/world/area_kkj/kkj_14/ADAEF0.c [hunk 8]
```diff
@@ -1,17 +1,44 @@
...
+    gDPSetPrimColor(gMasterGfxPos++, 0, 0, 0, 0, 0, D_80241D3B_ADCBDB);
+}
+
+ApiStatus func_80240480_ADB320(Evt* script, s32 isInitialCall) {
+    gGameStatusPtr->peachFlags |= 0x10;
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_8024049C_ADB33C(Evt* script, s32 isInitialCall) {
+    gGameStatusPtr->peachFlags &= 0xEF;
```
Comment (ethteck) [inline src/world/area_kkj/kkj_14/ADAEF0.c:42]: ~0x10

File: src/world/area_kkj/kkj_15/AE5590.c [hunk 9]
```diff
@@ -9,4 +9,9 @@ static char* N(exit_str_5) = "";
 
 #include "world/common/atomic/TexturePan.inc.c"
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_15/AE5590", func_8024030C_AE589C);
+ApiStatus func_8024030C_AE589C(Evt* script, s32 isInitialCall) {
+    gGameStatusPtr->peachFlags &= 0xFE;
```
Comment (ethteck) [inline src/world/area_kkj/kkj_15/AE5590.c:13]: ~0x1

File: src/world/area_kkj/kkj_18/AF2FE0.c [hunk 10]
```diff
@@ -1,17 +1,108 @@
...
+    Camera* camera = &gCameras[CAM_DEFAULT];
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_18/AF2FE0", func_802405F8_AF3298);
+    if (isInitialCall) {
+        camera->flags |= CAM_FLAG_SHAKING;
+        script->functionTempF[0] = 0.0f;
+    }
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_18/AF2FE0", func_80240694_AF3334);
+    script->functionTempF[0] = script->functionTempF[0] + 10.0f;
```
Comment (ethteck) [inline src/world/area_kkj/kkj_18/AF2FE0.c:100]:
```suggestion
    script->functionTempF[0] += 10.0f;
```

File: src/world/area_kkj/kkj_26/B13120.c [hunk 11]
```diff
@@ -5,4 +5,9 @@ static char* N(exit_str_1) = "kkj_14";
 static char* N(exit_str_2) = "kkj_14\0";
 static char* N(exit_str_3) = "kkj_26";
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_26/B13120", func_80240000_B13120);
+ApiStatus func_80240000_B13120(Evt* script, s32 isInitialCall) {
+    gGameStatusPtr->peachFlags &= 0xFE;
```
Comment (ethteck) [inline src/world/area_kkj/kkj_26/B13120.c:9]: ~0x1

File: src/world/area_kkj/kkj_29/B150B0.c [hunk 12]
```diff
@@ -3,4 +3,9 @@
 static char* N(exit_str_0) = "jan_22";
 static char* N(exit_str_1) = "kkj_10";
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_29/B150B0", func_80240000_B150B0);
+ApiStatus func_80240000_B150B0(Evt* script, s32 isInitialCall) {
+    gGameStatusPtr->peachFlags &= 0xFE;
```
Comment (ethteck) [inline src/world/area_kkj/kkj_29/B150B0.c:7]: again

**Review by ethteck — CHANGES_REQUESTED** [review]
great stuff here!

File: src/world/area_kkj/kkj_18/AF2FE0.c [hunk 13]
```diff
@@ -1,17 +1,108 @@
...
+    if (ptr != NULL) {
+        for (i = 0; ptr[i] != 0; i++) {
+            tik_07_FlyingAI_JumpVels[i] = ptr[i];
+        }
+        tik_07_FlyingAI_JumpVels[i] = 0;
+    } else {
+        for (i = 0; i < 112; i++) {
+            tik_07_FlyingAI_JumpVels[i] = i + 16;
+            tik_07_FlyingAI_JumpVels[112] = 0;
+        }
```
Comment (mrshigure) [inline src/world/area_kkj/kkj_18/AF2FE0.c:57]: In cases like this, where we don't know the size of the array, would it be reasonable to assume `ARRAY_COUNT(tik_07_FlyingAI_JumpVels) = 112` and increase it later if needed?

File: src/world/area_kkj/kkj_14/ADAEA0.c (hunk 6 above)
Comment (mrshigure) [reply src/world/area_kkj/kkj_14/ADAEA0.c:7]: I'm assuming you're talking about those `static char* N(exit_str_x)` strings, is that right? I tried to use this pattern in a couple of ways. The function matches but the produced assembly fails somewhere. I think I just don't have a good grasp of how things are connected yet. Are you suggesting something like this:
```c
ApiStatus func_80240000_ADAEA0(Evt* script, s32 isInitialCall) {
    if (evt_get_variable(NULL, GSW(0)) == -128) {
        sprintf(D_800D9668, "nok_bg");
    }
    return ApiStatus_BLOCK;
}

static char* N(exit_str_0) = "\0pra_01\0\0kkj_11\0\0kkj_01\0\0kkj_26\0\0kkj_15\0\0kkj_27\0\0kkj_26";
```

Comment (ethteck) [reply src/world/area_kkj/kkj_14/ADAEA0.c:7]: yeah, exactly that! I recommend using vbindiff when solving issues like these, but it can be tricky. I'm happy to try looking on your branch as well if you want

Comment (ethteck) [reply src/world/area_kkj/kkj_14/ADAEA0.c:7]: ah, I just looked and you have it correct - you just need to remove the leading \0 from your string below the function, and it matches :) we can merge once this is done

Comment (mrshigure) [reply src/world/area_kkj/kkj_14/ADAEA0.c]: Thanks for looking! I was confident I had tested this possibility. PR ready to be merged!
